From 090ac46912025f94aca710355e51a8154df36a56 Mon Sep 17 00:00:00 2001
From: Niccol Venerandi <niccolo@venerandi.com>
Date: Wed, 11 Mar 2020 18:46:07 +0100
Subject: [PATCH 1/2] Added support for libgit>=.99

Summary: LIBGIT2_SOVERSION no longer returns an integer in libgit>=.99; this adds support for that version and 1.0 to come.

Differential Revision: https://phabricator.kde.org/D27990


From 9a02338bfa60336b7349ff8ac1f85793515ff0e3 Mon Sep 17 00:00:00 2001
From: Niccol Venerandi <niccolo@venerandi.com>
Date: Thu, 12 Mar 2020 10:53:05 +0100
Subject: [PATCH 2/2] LIBGIT_VER_MINOR is supposed to be LIBGIT2_VER_MINOR

Reviewers: ognarb

Reviewed By: ognarb

Differential Revision: https://phabricator.kde.org/D28001
---
 src/application.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/application.cpp b/src/application.cpp
index c52fb57..b673848 100644
--- a/src/application.cpp
+++ b/src/application.cpp
@@ -56,7 +56,7 @@ Application::Application(int &argc, char **argv)
 
 
     #ifdef WITH_LIBGIT2
-        #if LIBGIT2_SOVERSION >= 22
+        #if LIBGIT2_VER_MAJOR > 0 || (LIBGIT2_VER_MAJOR == 0 && LIBGIT2_VER_MINOR >= 22)
             git_libgit2_init();
         #else
             git_threads_init();
@@ -67,7 +67,7 @@ Application::Application(int &argc, char **argv)
 Application::~Application()
 {
     #ifdef WITH_LIBGIT2
-        #if LIBGIT2_SOVERSION >= 22
+        #if LIBGIT2_VER_MAJOR > 0 || (LIBGIT2_VER_MAJOR == 0 && LIBGIT2_VER_MINOR >= 22)
             git_libgit2_shutdown();
         #else
             git_threads_shutdown();
-- 
2.26.0
